src/github.com/couchbaselabs/sync_gateway/base/dcp_feed.go

```diff
@@ -0,0 +1,193 @@
+//  Copyright (c) 2014 Couchbase, Inc.
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package base
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/couchbase/gomemcached"
+	"github.com/couchbaselabs/go-couchbase/cbdatasource"
+	"github.com/couchbaselabs/walrus"
+)
+
+type couchbaseDCPFeedImpl struct {
+	bds    cbdatasource.BucketDataSource
+	events <-chan walrus.TapEvent
+}
+
+func (feed *couchbaseDCPFeedImpl) Events() <-chan walrus.TapEvent {
+	return feed.events
+}
+
+func (feed *couchbaseDCPFeedImpl) Close() error {
+	return feed.bds.Close()
+}
+
+// Implementation of couchbase.AuthHandler for use by cbdatasource during bucket connect.
+type dcpAuth struct {
+	Username, Password, BucketName string
+}
+
+// GetCredentials is based on approach used in go-couchbase pools.go authHandler().
+func (a dcpAuth) GetCredentials() (string, string, string) {
+	if a.Username == "" {
+		return a.BucketName, "", a.BucketName
```

Comment:
After hitting an authentication issue (prior to making this change) when creating a new bucket with the standard username/password authenticator, I talked to Steve Yen and he tipped me off about the default bucketname-as-username handling. Since it wasn't happening when using the go-couchbase bucket, I went digging in their code to see where they were handling it, and found it in the go-couchbase authHandler() method I mentioned in the comment:

```
func (b Bucket) authHandler() (ah AuthHandler) {
	if b.pool != nil {
		ah = b.pool.client.ah
	}
	if ah == nil {
		ah = &basicAuth{b.Name, ""}
	}
	return
}
```
I talked to Steve about whether there's any official documentation for the changes in the Sherlock CB API auth handling (which this is related to), and he said there isn't anything yet - all the teams are just adjusting to the auth changes as they come.  I think we're in the same boat for now.